de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/followmode/FollowModeNotificationDispatcher.java

```diff
@@ -0,0 +1,66 @@
+package de.fu_berlin.inf.dpp.intellij.followmode;
+
+import de.fu_berlin.inf.dpp.editor.FollowModeManager;
+import de.fu_berlin.inf.dpp.editor.IFollowModeListener;
+import de.fu_berlin.inf.dpp.intellij.ui.Messages;
+import de.fu_berlin.inf.dpp.intellij.ui.util.NotificationPanel;
+import de.fu_berlin.inf.dpp.session.User;
+import org.apache.log4j.Logger;
+
+public class FollowModeNotificationDispatcher implements IFollowModeListener {
+
+  private static final Logger log = Logger.getLogger(FollowModeNotificationDispatcher.class);
+
+  private volatile User followedUser;
+
+  public FollowModeNotificationDispatcher(FollowModeManager followModeManager) {
+    followModeManager.addListener(this);
+  }
+
+  @Override
+  public void stoppedFollowing(Reason reason) {
+    String shownReason;
+
+    switch (reason) {
+      case FOLLOWEE_LEFT_SESSION:
+        shownReason = Messages.FollowModeNotificationDispatcher_end_reason_FOLLOWEE_LEFT_SESSION;
+        break;
+      case FOLLOWER_CLOSED_OR_SWITCHED_EDITOR:
+        shownReason =
+            Messages.FollowModeNotificationDispatcher_end_reason_FOLLOWER_CLOSED_OR_SWITCHED_EDITOR;
+        break;
+      case FOLLOWER_CLOSED_EDITOR:
+        shownReason = Messages.FollowModeNotificationDispatcher_end_reason_FOLLOWER_CLOSED_EDITOR;
+        break;
+      case FOLLOWER_STOPPED:
+        shownReason = Messages.FollowModeNotificationDispatcher_end_reason_FOLLOWER_STOPPED;
+        break;
+      case FOLLOWER_SWITCHES_FOLLOWEE:
+        shownReason =
+            Messages.FollowModeNotificationDispatcher_end_reason_FOLLOWER_SWITCHES_FOLLOWEE;
+        break;
+
+      default:
+        shownReason = reason.name();
+        log.error("Encountered unknown reason for follow mode end. reason: " + reason);
```

Comment:
Warning should be enough at it is only display logic and not something critical.

Reply:
Done.